typescript/introduction/intro-typescript/what-is-typescript.md

```diff
@@ -0,0 +1,47 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Intro to TypeScript
+
+---
+## Content
+
+If you're old enough, you might remember when JavaScript was just a web development language. Due to its versatility, JavaScript is now used in many applications including neural network design, power backends and many more.
+
+You're probably thinking "What's JavaScript got to do with it -- isn't this a TypeScript course?". Let us explain!
+
+Because of the high adoption rate of JavaScript, many tools (such as Angular, React, Node.js, etc) were created to allow programmers to write better/safer JavaScript code. 
+
+TypeScript is another creation meant to put the future of JavaScript in the hands of developers, and it's actually a **superset of JavaScript**. What that means is that TypeScript has all the functionality of JavaScript, but *adds even more*.
+
+![typescript-is-superset-of-javascript](https://img.enkipro.com/074c0352be386e58bbdd697ed8bff427.png)
+
+---
+## Practice
+
+Any valid JavaScript is also valid TypeScript.
+
+???
+
+* True.
```

Comment:
https://www.typescriptlang.org/play/#code/DYUwLgBAhhC8EHIASBLBBuAUDeBGX6QA

Reply:
That's just a matter of compiler flags.

Syntactically, every JS file is a valid TS file, although compilation might produce errors for some, depending on the compiler config flags.

[Here's a good explanation](https://stackoverflow.com/a/44967932)

Reply:
That being said, I just this could still lead to confusion.

Maybe we should say:

> Any valid JavaScript syntax is also valid TypeScript syntax

Although then we'd also need to introduce what "syntax" is.

Reply:
either way, I don't think we should be this vocal about this detail - I think there are many nuances to the statement which we can't properly describe unless we go into the details of the language:

a) in the snippet you shared they talk about programs and not valid programs; isn't this a big difference?
b) we have a seeming counterexample down below: https://github.com/enkidevs/curriculum/pull/1819/files/78e43d2ba6a665799a5199ef305b2f52aa85103a#diff-c4113eb1d2839c3f32285073c91d3fa1
c) we don't mention anything about compiler flags - if we just assume the default ones then the first impression is that js programs directly migrated to typescript can fail and throw errors -> that makes the program invalid?

how would we phrase this nuance? typescript can raise errors, but if we really want to run the compiled program it will work? or is it valid from JS pov, but not TS pov?
at which point is a program a typescript program - post compilation? doesn't make sense

Reply:
to make that statement true we have to say something like:
- well typescript errors can refer to a) errors that make the program invalid b) errors about the typing of the code, but don't make your program invalid 

if you program only has b) then your program can still be run by typescript

I can't stop thinking that this will just confuse users

Reply:
Agreed.